internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -181,28 +181,44 @@ func (r *PreprovisioningImageReconciler) Reconcile(origCtx context.Context, req
 		return ctrl.Result{Requeue: true, RequeueAfter: time.Until(imageTimePlusCooldown)}, nil
 	}
 
-	patch := client.MergeFrom(image.DeepCopy())
-	imageUpdated := image.Status.ImageUrl != "" && image.Status.ImageUrl != infraEnv.Status.ISODownloadURL
-	err = r.setImage(image, *infraEnv)
+	return ctrl.Result{}, r.handleImageUpdate(ctx, log, image, infraEnv)
+}
+
+func (r *PreprovisioningImageReconciler) handleImageUpdate(ctx context.Context, log logrus.FieldLogger, image *metal3_v1alpha1.PreprovisioningImage, infraEnv *aiv1beta1.InfraEnv) error {
+	bmh, err := r.getBMH(ctx, image)
 	if err != nil {
-		return ctrl.Result{}, err
+		return err
+	}
+
+	patch := client.MergeFrom(image.DeepCopy())
+	imageUpdated := false
+	if bmh.Status.Provisioning.State == metal3_v1alpha1.StateProvisioned {
+		log.Info("Ignoring image update for provisioned BMH")
```

Comment:
This might effect Agent return to pool.
Consider this scenario:
1. Host is installed successfully.
2. InfraEnv get updated (while the host is already provisioned) - no update to the PPI
3. Cluster is decommissioned / host removed from the cluster -> ClusterDeployment Ref removed from the agent - > the host should reboot into discovery ISO which is - the old outdated image in the PPI .

Reply:
My assumption here was that the BMH will change state and at that point we'll update the preprovisioning image.

I'll look into this manually and see if it's working.

Ideally whatever we do to deprovision and reprovision the BMH will cause this to get updated, but we'll want to be sure we don't have a race.

I could allow the url to be updated for provisioned BMHs and just only set the reboot annotation for non-provisioned ones. Will have to think if that could break anything though.